internal/config/context.go

```diff
@@ -1,54 +1,126 @@
 package config
 
 import (
+	"errors"
+	"fmt"
 	"os"
 
 	"github.com/Ensono/taskctl/pkg/runner"
 	"github.com/Ensono/taskctl/pkg/variables"
+	"github.com/sirupsen/logrus"
 
-	"github.com/Ensono/taskctl/pkg/utils"
+	"github.com/Ensono/taskctl/internal/utils"
 )
 
+var defautlContainerExcludes = []string{"PATH", "HOME", "TMPDIR"}
+
+var ErrBuildContextIncorrect = errors.New("build context properties are incorrect")
+
 func buildContext(def *ContextDefinition) (*runner.ExecutionContext, error) {
 	dir := def.Dir
 	if dir == "" {
 		dir = utils.MustGetwd()
 	}
-	if def.Envfile != nil {
-		def.Envfile = utils.NewEnvFile(func(e *utils.Envfile) {
-			e.Generate = def.Envfile.Generate
-			e.Delay = def.Envfile.Delay
-			e.Exclude = def.Envfile.Exclude
-			e.Include = def.Envfile.Include
-			// e.Path = def.Envfile.Path
-			e.Modify = def.Envfile.Modify
-			e.Quote = def.Envfile.Quote
-			e.ReplaceChar = def.Envfile.ReplaceChar
-		})
-		if err := def.Envfile.Validate(); err != nil {
-			return nil, err
-		}
+	if def.Container != nil && def.Container.Name == "" {
+		return nil, fmt.Errorf("either container image must be specified, %w", ErrBuildContextIncorrect)
+	}
+
+	if def.Executable != nil && def.Executable.Bin == "" {
+		return nil, fmt.Errorf("executable binary must be specified, %w", ErrBuildContextIncorrect)
 	}
 
 	osEnvVars := variables.FromMap(utils.ConvertFromEnv(os.Environ()))
 	userEnvVars := variables.FromMap(def.Env)
-
 	// build an env order is _IMPORTANT_
 	// we need to overwrite any existing env vars with the user supplied ones
 	buildEnvVars := osEnvVars.Merge(userEnvVars)
+	envFile, err := newEnvFile(def.Envfile, def.Container != nil)
+	if err != nil {
+		return nil, err
+	}
 
 	c := runner.NewExecutionContext(
-		def.Executable,
+		contextExecutable(def),
 		dir,
 		buildEnvVars,
-		def.Envfile,
+		envFile,
 		def.Up,
 		def.Down,
 		def.Before,
 		def.After,
-		runner.WithQuote(def.Quote),
+		runner.WithQuote(def.Quote), func(c *runner.ExecutionContext) {
+			c.Variables = variables.FromMap(def.Variables)
+		},
 	)
-	c.Variables = variables.FromMap(def.Variables)
-
 	return c, nil
 }
+
+func newEnvFile(defEnvFile *utils.Envfile, isContainerContext bool) (*utils.Envfile, error) {
+	if defEnvFile == nil && !isContainerContext {
+		return defEnvFile, nil
+	}
+
+	envFile := utils.NewEnvFile(func(e *utils.Envfile) {
+		e.Generate = defEnvFile.Generate
+		if isContainerContext {
+			e.Generate = true
+		}
+		e.Delay = defEnvFile.Delay
```

Comment:
We can remove the delay feature, it never fixed what it was meant to

Reply:
:D I made this note in the code 
![image](https://github.com/user-attachments/assets/57d47f00-00af-40d2-bc13-30962702fc63)